source/index.ts

```diff
@@ -11,6 +9,11 @@ const validFilename = require('valid-filename')
  * @param {Array<string>} invalidPaths - if validation failed, this an array of the full paths that failed
  * @returns {void}
  */
+type validateCallback = (
+	error: Error | null,
+	valid?: boolean,
+	invalidPaths?: string[]
+) => void
```

Comment:
we should convert errbacks (callbacks which first param is an error) to promises when we do our typescript conversions

I can show you how to do the promise conversion as the next major release after this is merged